controllers/imagepolicy_controller.go

```diff
@@ -284,3 +307,42 @@ func (r *ImagePolicyReconciler) patchStatus(ctx context.Context, req ctrl.Reques
 
 	return r.Status().Patch(ctx, &res, patch)
 }
+
+func (r *ImagePolicyReconciler) hasAccessToRepository(ctx context.Context, policy ctrl.Request, repo meta.NamespacedObjectReference, acl *imagev1.AccessFrom) (bool, error) {
+	// grant access if the policy is in the same namespace as the repository
+	if repo.Namespace == "" || policy.Namespace == repo.Namespace {
+		return true, nil
+	}
+
+	// deny access if the repository has no ACL defined
+	if acl == nil {
+		return false, fmt.Errorf("ImageRepository '%s/%s' can't be accessed due to missing access list",
+			repo.Namespace, repo.Name)
+	}
+
+	// grant access if the repository ACL has no namespace selectors
+	if acl != nil && acl.NamespaceSelectors == nil {
```

Comment:
I still think this is odd -- no selectors should mean no access (because none can match). I would just remove this special case.

Reply:
Hmm but `accessFrom: {}` should allow all right?

Reply:
Also this should allow all:
```yaml
  accessFrom:
    namespaceSelectors:
      - matchLabels: {}
```

Reply:
That should read as "access from nothing", I would think. Think of it like this: the contents of `accessFrom` specify all the circumstances in which access is allowed. An empty value (nil or empty list) means no circumstances.
In contrast, the contents of `matchLabels` gives all the labels that must match -- if an empty map, there are no requirements (i.e., any set of labels trivially matches).

Reply:
Ok so to allow access from all namespaces, we'll only support `matchLabels: {}`?

Reply:
One fiddliness is whether `accessFrom: {}` and `accessFrom: {namespaceSelectors: nil}` mean no access whatsoever, or access only from the same namespace.

Reply:
I think I covered the empty ACL (deny) and empty labels (allow) with tests in latest commit.